core/templates/domain/state/StateObjectFactory.ts

```diff
@@ -109,6 +111,40 @@ export class State {
     this.writtenTranslations = otherState.writtenTranslations;
     this.nextContentIdIndex = otherState.nextContentIdIndex;
   }
+
+  getRequiredWrittenTranslationsCount(): number {
+    let interactionId = this.interaction.id;
+    const contentIdsToRemove = [];
+
+    let allContentIds = this.writtenTranslations.getAllContentId();
```

Comment:
Maybe using a set and just removing directly might be faster? Otherwise it's a fair number of comparisons -- seems inefficient.